crates/bevy_ecs/src/query/state.rs

```diff
@@ -74,6 +74,175 @@ pub struct QueryState<D: QueryData, F: QueryFilter = ()> {
     par_iter_span: Span,
 }
 
+/// View into a [`QueryState`] to avoid split-borrow issues when using [`QueryState`]
+pub struct QueryStateView<'me, D: QueryData, F: QueryFilter = ()> {
```

Comment:
Why `'me` out of curiosity?

Reply:
Because I wanted to convey that this was a reborrow of `&QueryState` with the same lifetime. Maybe that makes it confusing. I could change it to `'self`' or `'a`